fpdcc/document-search
# Send license expiration notification emails

## Overview

This creates a management command `send_expiration_email` that puts details for all licenses expiring within a year of the command's start date into an email, which will be sent to users in the `NotificationSubscription` model. The intention is that this will be run once a year with a cron job.

There is a fix here for an error with validators that would not allow for migrations to be made. The fix adds `@wraps` decorators to certain validator functions.

This also adds an `.env.example` file intended to store the mandrill credentials and adds more context to `local_settings.example.py` for email sending.

Closes #106
Connects #63 

### Notes

For testing, my local db didn't have any licenses that were expiring this year, so I brought the app up and made one.

Also, this does not include work done to make a cron job. To my understanding, it seems like that's done on the deployment instance.

## Testing Instructions

* Copy the contents of `.env.example` to a new `.env` file and plug in the mandrill api key
* Run migrations for the new model and validators
  * [Load in initial data if needed](https://github.com/fpdcc/document-search#loading-initial-data)
* Log in to the admin and add one or two users to `NotificationSubscription`
* Run `docker-compose run --rm app python manage.py send_expiration_email`
* Confirm that appropriate statuses display in the terminal and all users associated with notifications were sent an email
* Confirm that:
  * if there are licenses expiring, the email says how many and includes a csv with appropriate details
  * if there are no licenses expiring, the email says that, and does not have any attachments
